test/_functions/python/termination-hook/termination-hook.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 The Nuclio Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+import os
+
+
+class TerminationHandler:
+    file_path = '/tmp/nuclio/termination-hook.txt'
+
+    def __init__(self, worker_id, logger):
+        self.worker_id = worker_id
+        self.logger = logger
+
+    def write_results(self):
+
+        # if the file doesn't exist, create it
+        if not os.path.exists(self.file_path):
+            self.logger.info_with('Creating file', file_path=self.file_path)
+            os.makedirs(os.path.dirname(self.file_path), exist_ok=True)
+            open(self.file_path, 'w').close()
+
+        with open(self.file_path, 'a') as f:
+            self.logger.info_with('Writing to file', file_path=self.file_path)
+            f.write(f'Worker {self.worker_id} - Done!\n')
```

Comment:
I know this is just for testing, but multiple workers may append concurrently and corrupt this file. There may even be a race condition where workers truncate the file after writing has already started by another worker. Seems to me that the worker ID needs to be in the file name.